/*
 *  Copyright (C) 2023 â€“ 2024 Advanced Micro Devices, Inc. All rights reserved.
 *  Licensed under the MIT License.
 */

syntax = "proto3";
package vaip_core;
import "anchor_point.proto";

message DpuSubgraphReportProto {
  repeated DpuSubgraphEntryProto subgraphs = 1;
}

message XirTensorProto {
  string name = 1;
  repeated int64 shape = 2;
}

message TryFuseProto {
  repeated string inputs = 1;
  repeated string outputs = 2;
  repeated string arguments = 3;
  repeated string return_values = 4;
  repeated string body = 5;
}

message DpuSubgraphEntryProto {
  string status = 1; // "OK"
  string subgrpah_name = 2;
  repeated XirTensorProto xir_subgraph_inputs  = 3;
  repeated XirTensorProto xir_subgraph_outputs  = 4;
  repeated AnchorPointProto xir_input_anchor_points = 5;
  repeated AnchorPointProto xir_output_anchor_points = 6;
  TryFuseProto try_fuse = 7;
  repeated AnchorPointProto onnx_output_anchor_points = 8;
  string comments = 100;
  repeated string try_fuse_loop_path = 101;
}
